Sources/Fluent/Database/Database.swift

```diff
@@ -1,21 +1,23 @@
-import Foundation
-
 /**
     References a database with a single `Driver`.
     Statically maps `Model`s to `Database`s.
 */
-public class Database {
-    /**
-        The `Driver` powering this database. 
-        Responsible for executing queries.
-    */
+public class Database: Executor {
+    /// The `Driver` powering this database.
+    /// Responsible for executing queries.
     public let driver: Driver
+    
+    /// Maintains a pool of connections
+    /// one for each thread
+    public let threadConnectionPool: ThreadConnectionPool
 
-    /** 
-        Creates a `Database` with the supplied
-        `Driver`. This cannot be changed later.
-    */
+    /// Creates a `Database` with the supplied
+    /// `Driver`. This cannot be changed later.
     public init(_ driver: Driver) {
+        threadConnectionPool = ThreadConnectionPool(
+            makeConnection: driver.makeConnection,
+            maxConnections: 5
```

Comment:
Can we make the `maxConnections` have a default value but be a configurable option to be set elsewhere? That would allow it to be changed to fit the infrastructure specs.